src/go/rdctl/cmd/extensionInstall.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright © 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cmd implements the rdctl commands
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+)
+
+var forceInstall bool
+
+// installCmd represents the 'rdctl extensions install' command
+var installCmd = &cobra.Command{
+	Use:   "install",
+	Short: "Install an RDX extension",
+	Long: `rdctl extension install [--force] <image-id>
+--force: avoid any interactivity.
+The <image-id> is an image reference, e.g. splatform/epinio-docker-desktop:latest (the tag is optional).`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return fmt.Errorf("no image specified")
+		}
+		if len(args) >= 2 {
+			return fmt.Errorf("too many arguments specified")
+		}
```

Comment:
Why are you not letting cobra check the arguments?

```suggestion
    Args: cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
```

Reply:
Because it doesn't do a good job:

```
# Pull out the arg-checking code

$ go build -ldflags '-s -w' -o ../../../resources/darwin/bin/rdctl .
$ rdctl extension install
panic: runtime error: index out of range [0] with length 0
[... stacktrace omitted]
$ rdctl extension install abc def ghi
Error: installation failed: The image abc has invalid extension metadata
$ 
```

Reply:
I gave a comprehensive answer for this, and it's lost. Rather than put in all the details, I'll sum up:

1. I forgot about them

2. `cobra.ExactArgs(1)` output is ok
3. `cobra.NoArgs` output for `rdctl extension ls abc def` is confusing, so I didn't replace the custom code.